drivers/Aqara/aqara-presence-sensor/src/fp2/device_manager.lua

```diff
@@ -0,0 +1,272 @@
+local log = require "log"
+local json = require "st.json"
+local fields = require "fields"
+local capabilities = require "st.capabilities"
+local multipleZonePresence = require "multipleZonePresence"
+
+local PresenceSensor = capabilities.presenceSensor
+local MovementSensor = capabilities.movementSensor
+local ActivitySensor = capabilities.activitySensor
+local DeviceMode = capabilities["stse.deviceMode"]
+
+local MOVEMENT_TIMER = "movement_timer"
+local MOVEMENT_TIME = 5
+local COMP_MODE = "mode"
+
+local device_manager = {}
+device_manager.__index = device_manager
+
+local FP2_MODES = { "zoneDetection", "fallDetection", "sleepMonitoring" }
+
+function device_manager.presence_handler(driver, device, zone, evt_value)
+  if not device:supports_capability(PresenceSensor) then return end
+  local evt_action = "not present"
+  if evt_value == 1 then evt_action = "present" end
+  device:emit_event(PresenceSensor.presence(evt_action))
+end
+
+function device_manager.movement_handler(driver, device, zone, evt_value)
+  if not device:supports_capability(PresenceSensor) then return end
+
+  local val = evt_value
+  local no_movement = function()
+    if not device:supports_capability(PresenceSensor) then return end
```

Comment:
Should be checking for `MovementSensor` here as well

Reply:
Okay. You're right.